libs/s25main/random/Random.h

```diff
@@ -139,3 +140,45 @@ class RandomFunctor
 
 /// Shortcut for creating an instance of RandomFunctor
 #define RANDOM_SHUFFLE(container) RandomFunctor::shuffleContainer(container, __FILE__, __LINE__)
+
+/// Return a random number in the interval [lowerBound, upperBound].
+template<typename RandomT>
+static unsigned getRandomNumber(const RandomT& rng, unsigned lowerBound, unsigned upperBound)
+{
+    std::uniform_int_distribution<typename RandomT::result_type> dist(lowerBound, upperBound);
+    return dist(rng);
+}
+
+/// Return a random number in the interval [0, upperBound).
+template<typename RandomT>
+static unsigned getRandomNumber(const RandomT& rng, unsigned upperBound)
+{
+    RTTR_Assert(upperBound > 0);
+    return getRandomNumber(rng, 0, upperBound - 1);
```

Comment:
Better: 
```suggestion
inline unsigned getRandomIndex(const RandomT& rng, unsigned size)
{
    RTTR_Assert(size > 0);
    return getRandomNumber(rng, 0, size - 1);
```

Reply:
> See comments. Mostly about inline and non-const rng.
> Also a suggestion about not naming a non-inclusive range "upperBound". I think this makes it clearer

- const rng was an oversight.
- inline seems to be counter productive. What is the reason you want it to have inline linkage over internal linkage?

Reply:
Inline mainly because it is in a header. If it was a cpp file, then yes, internal linkage but this way it would be internal to any cpp file which includes this header which isn't what one usually wants. Besides that the compiler warns about unused static functions but not for inline functions, so that's what I'd use